sympy/functions/combinatorial/factorials.py

```diff
@@ -339,13 +339,14 @@ def eval(cls, arg):
                 if arg.is_even:
                     k = arg / 2
                     return 2 ** k * factorial(k)
+                if arg.is_odd:
+                    return factorial(arg) / factorial2(arg - 1)
+                raise ValueError("argument must be nonnegative integer or odd integer")
```

Comment:
@jksuom Thanks for your help! So, I guess the code is more efficient now, than the one was previous. So, Do I need to add anything more now?